File: packages/backend/src/common/types/trade-execution-result.ts
export type TradeExecutionStatus = 'filled' | 'rejected' | 'partial' | 'error';

export interface TradeExecutionResult {
  id: string;                    // UUID
  signal_id: string;
  account_id: string;
  order_id: number;              // Broker order ID
  fill_price: number;
  execution_latency_ms: number;
  status: TradeExecutionStatus;
  rejection_reason?: string;
  slippage: number;
  spread_at_execution: number;
  created_at: string;            // ISO 8601
}

File: packages/execution-engine/src/models/trade_execution_result.py
from enum import Enum
from typing import Optional

from pydantic import BaseModel


class TradeExecutionStatus(str, Enum):
    FILLED = "filled"
    REJECTED = "rejected"
    PARTIAL = "partial"
    ERROR = "error"


class TradeExecutionResult(BaseModel):
    id: str  # UUID
    signal_id: str
    account_id: str
    order_id: int  # Broker order ID
    fill_price: float
    execution_latency_ms: float
    status: TradeExecutionStatus
    rejection_reason: Optional[str] = None
    slippage: float
    spread_at_execution: float
    created_at: str  # ISO 8601
